cranelift/codegen/src/legalizer/boundary.rs

```diff
@@ -113,12 +114,33 @@ fn legalize_entry_params(func: &mut Function, entry: Block) {
 
         let abi_type = pos.func.signature.params[abi_arg];
         let arg_type = pos.func.dfg.value_type(arg);
+        match abi_type.purpose {
+            ArgumentPurpose::StructArgument(size) => {
+                let offset = if let ArgumentLoc::Stack(offset) = abi_type.location {
+                    offset
+                } else {
+                    unreachable!("StructArgument must already have a Stack ArgumentLoc assigned");
+                };
+                let ss = pos.func.stack_slots.make_incoming_arg(size, offset);
+                let struct_arg = pos.ins().stack_addr(arg_type, ss, 0);
+                pos.func.dfg.change_to_alias(arg, struct_arg);
+                let dummy = pos
+                    .func
+                    .dfg
+                    .append_block_param(entry, crate::ir::types::SARG_T);
+                pos.func.locations[dummy] = ValueLoc::Stack(ss);
+                abi_arg += 1;
+                continue;
+            }
+            _ => {}
+        }
```

Comment:
uber nit: can you add a \n after this match (or `if`, if you agree with the above proposal)